packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx

```diff
@@ -15,21 +15,29 @@ export const ResponsiveGridRoot = styled(Grid.Root)`
   container-type: inline-size;
 `;
 
-export const ResponsiveGridItem = styled(Grid.Item)`
-  grid-column: span 12;
-
-  @container (min-width: ${RESPONSIVE_CONTAINER_BREAKPOINTS.sm}) {
-    ${({ col }) => col && `grid-column: span ${col};`}
-  }
-`;
+// We need to use a different grid item for the responsive layout in the test environment
+// because @container is not supported in jsdom and it throws an error
+export const ResponsiveGridItem =
+  process.env.NODE_ENV !== 'test'
+    ? styled(Grid.Item)<{ col: number }>`
+        grid-column: span 12;
+        @container (min-width: ${RESPONSIVE_CONTAINER_BREAKPOINTS.sm}) {
+          ${({ col }) => col && `grid-column: span ${col};`}
+        }
+      `
+    : styled(Grid.Item)<{ col: number }>`
+        grid-column: span 12;
+      `;
 
 interface FormLayoutProps extends Pick<EditLayout, 'layout'> {
   hasBackground?: boolean;
+  model?: string;
 }
 
-const FormLayout = ({ layout, hasBackground = false }: FormLayoutProps) => {
+const FormLayout = ({ layout, hasBackground = true, model: modelContext }: FormLayoutProps) => {
   const { formatMessage } = useIntl();
-  const { model } = useDoc();
+  const { model: modelHook } = useDoc();
+  const model = modelContext || modelHook;
```

Comment:
To be sure I understand what's going on. For a relation modal, useDoc will get a match, but not the one from the relation modal, instead a match from the preview or edit view page underneath.

But if we have the model passed in the props, which is the case for relations modal, we use that in priority.

Is that right? It would be nice to have a comment somewhere to explain why we're doing this

Reply:
yes that's right, I can add a comment, sure